### PR TITLE
Use current journey plan for LiveRide replans

### LINKED ISSUE #476: Use current route type for replan
When re-planning a route from LiveRide, use the current route type rather than the one in settings.
e.g. I might have "balanced" as preferred type in my settings, but I might decide I want to use the fastest route when riding a particular route.  If I go off route then I'd like the replan to continue to be the fastest route rather than changing it to balanced.

### PR DESCRIPTION
**Pull request by HilaryN**
Fixes #476 

[Test LiveRide replan uses current plan.pdf](https://github.com/cyclestreets/android/files/6940526/Test.LiveRide.replan.uses.current.plan.pdf)

### CODE REVIEW

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Route.kt [hunk 1]
```diff
@@ -50,11 +52,17 @@ object Route {
...
+    fun LiveReplanRoute(speed: Int,
                         context: Context,
                         waypoints: Waypoints) {
-        val query = LiveRideReplanRoutingTask(plan, speed, context)
+        var newPlan: String
+        // Check current plan is a linear route plan.
+        if (currentJourneyPlan in RoutePlans.allPlans())
+            newPlan = currentJourneyPlan
+        else
+            newPlan = RoutePlans.PLAN_QUIETEST
```
Comment (oliverlockwood) [inline libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Route.kt:63]: As discussed I think this should be Balanced.

**Review by oliverlockwood — CHANGES_REQUESTED** [review]
I'm happy with this, subject to the suggested change.

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Route.kt (hunk 1 above)
Comment (HilaryN) [reply libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Route.kt:63]: I'm happy with Balanced, but it depends on whether Circular routes are consistently planned as quietest or balanced at the back end, in order to stick as closely as possible to the original route.
Hopefully @mvl22 can tell us that.

Comment (HilaryN) [reply libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Route.kt:63]: I made the change, then saw Martin's comment to say Circular routes are based on Quietest, so I've taken the change out again!
I've also changed Boolean to a primitive in RouteData.

**Review by oliverlockwood — APPROVED** [review]
Nice work.

**Pull request merged**